lib/api/exception.dart

```diff
@@ -15,9 +17,10 @@ sealed class ApiRequestException implements Exception {
   ///
   /// For [ZulipApiException] this is supplied by the server as the `message`
   /// property in the JSON response, and is translated into the user's language.
-  final String message;
+  final String Function(ZulipLocalizations) message;
```

Comment:
nit in commit message:

    core: Add translations for exceptions related to api requests

A better prefix would be `api:`. In particular "core" doesn't really mean anything here unless it's scoped within "api" (like the filename `lib/api/core.dart`) — there's lots of pieces of the codebase that could be thought of as "core".